x-pack/plugins/apm/public/components/shared/redirect_with_default_environment/index.tsx

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useLocation, useHistory } from 'react-router-dom';
+import qs from 'query-string';
+import React from 'react';
+import { last } from 'lodash';
+import { defaultApmServiceEnvironment } from '../../../../../observability/common';
+import { ENVIRONMENT_ALL } from '../../../../common/environment_filter_values';
+import { useApmPluginContext } from '../../../context/apm_plugin/use_apm_plugin_context';
+import { useApmRouter } from '../../../hooks/use_apm_router';
+
+export function RedirectWithDefaultEnvironment({
+  children,
+}: {
+  children: React.ReactElement;
+}) {
+  const { core } = useApmPluginContext();
+
+  const apmRouter = useApmRouter();
+  const location = useLocation();
+  const history = useHistory();
+
+  const query = qs.parse(location.search);
+
+  if ('environment' in query) {
+    return children;
+  }
+
+  const matchingRoutes = apmRouter.getRoutesToMatch(location.pathname);
+
+  if (last(matchingRoutes)?.path === '/services') {
```

Comment:
Why only check if the last matches?

Reply:
Would be nice if we had this with full type safety (the second argument should be checked against existing routes.

```ts
const isServiceInventory = apmRouter.isRoute(location.pathname, '/services')
```

Reply:
In this case, it's to ensure it only matches when /services is being rendered, and not one of its children. /services doesn't have children now, but when they are added, the redirect might be too aggressive. Maybe it's a little too defensive though, I'll see if there's a better way. I think a helper that takes a route path doesn't really help here, there could be multiple routes that have the same route path (e.g. `/`).